src/dsp_tools/analyse_xml_data/extract_links_from_XML.py

```diff
@@ -197,32 +240,32 @@ def analyse_circles_in_data(xml_filepath: Path, tracer_output_file: str) -> list
     print("=" * 80)
     tracer = VizTracer(
         minimize_memory=True,
-        ignore_c_function=True,
-        ignore_frozen=True,
-        include_files=["extract_links_from_XML.py", "models.py"],
+        max_stack_depth=3,
     )
```

Comment:
I think for now we have VizTracer only in the dev dependencies, so this might break when we ship it with only the prod dependencies. I think either we leave it in and add it to the dependencies, or we remove it entirely.

Reply:
I am happy with either. @jnussbaum what do you think?

Reply:
I thought this function won't be shipped anyway? But generally, I don't think that VizTracer should be in production code.

Reply:
How about moving this one function over to the testdata package, unlike the rest of this file, it's really only for us and doesn't need to be shipped. But that doesn't mean we have to change the function or even remove it; that just means it may want to live with the rest of the non-production code of the repo

Reply:
I also renamed the file, because it does not only extract the links but also analyses them.